numpy-stubs/__init__.pyi

```diff
@@ -993,6 +991,92 @@ def set_printoptions(
     *,
     legacy: Any = ...,
 ) -> None: ...
+def isscalar(element: Any) -> bool: ...
+def diagonal(a: _ArrayLike, offset: int = ..., axis1: int = ..., axis2: int = ...) -> ndarray: ...
+def allclose(
+    a: Union[_ArrayLike, _FloatLike],
+    b: Union[_ArrayLike, _FloatLike],
+    rtol: float = ...,
+    atol: float = ...,
+    equal_nan: bool = ...,
+) -> bool: ...
+
+#
+# ufunc
+#
+
+# Backported from latest NumPy
+class ufunc:
+    @property
+    def __name__(self) -> str: ...
+    def __call__(
+        self,
+        *args: Union[_FloatLike, _ArrayLike],
+        out: Optional[Union[ndarray, Tuple[ndarray, ...]]] = ...,
+        where: Optional[ndarray] = ...,
+        # The list should be a list of tuples of ints, but since we
+        # don't know the signature it would need to be
+        # Tuple[int, ...]. But, since List is invariant something like
+        # e.g. List[Tuple[int, int]] isn't a subtype of
+        # List[Tuple[int, ...]], so we can't type precisely here.
+        axes: List[Any] = ...,
+        axis: int = ...,
+        keepdims: bool = ...,
+        casting: Any = ...,
+        order: Any = ...,
+        dtype: Any = ...,
+        subok: bool = ...,
+        signature: Union[str, Tuple[str]] = ...,
+        # In reality this should be a length of list 3 containing an
+        # int, an int, and a callable, but there's no way to express
+        # that.
+        extobj: List[Union[int, Callable]] = ...,
+    ) -> Any: ...
```

Comment:
I don't love this Any return type. Is it possible to do an overload like we had before?

Reply:
I took this from [NumPy stubs](https://github.com/numpy/numpy/blob/ab585e6f8b047a9ca54578eb93e241917227e985/numpy/__init__.pyi#L2123), I guess the return type from ufuncs is too broad, check this issue: https://github.com/numpy/numpy/pull/17805